### ISSUE TITLE
Does this run single core?

### ISSUE THREAD

**Issue by Autumnlight02** [arcee-ai/mergekit#329]
So, for some reason its neither fully allocating my cpu, and if i run it using cuda it takes 8 hrs instead of 30min. is this tool only capable of running on a single core for merging models?

It is also somewhat inconsistent. Sometimes it goes really quickly and then it stops for a few seconds. Im confused.

**Comment by cg123** [comment]
Done in #331.

**Comment by cg123** [comment]
Yes, mergekit is currently single-threaded. PyTorch can do some internal threading on CPU - you may want to play with the OMP_NUM_THREADS environment variable.

The pauses you are seeing are probably shards of the output model being written to disk. I/O is a lot slower than the actual merge process.

**Comment by Autumnlight02** [comment]
Ahh, I wish there was a option to do it entirely in memory, I also run this on an nvme, so it should be rather fast
